staging/src/k8s.io/code-generator/pkg/util/build.go

```diff
@@ -50,3 +57,24 @@ func hasSubdir(root, dir string) (rel string, ok bool) {
 	// cut off root
 	return filepath.ToSlash(dir[len(root):]), true
 }
+
+// BoilerplatePath uses the boilerplate in k8s.io/code-generator.
+// If it does not exist, it uses the boilerplate in k8s.io/kubernetes.
+func BoilerplatePath() string {
+	boilerPlatePath := path.Join(reflect.TypeOf(empty{}).PkgPath(), "/../../hack/boilerplate.go.txt")
+	if ok, err := exists(filepath.Join(args.DefaultSourceTree(), "k8s.io/code-generator/hack/boilerplate.go.txt")); err == nil && ok {
```

Comment:
why the if clause? This means that the checked out version in the GOPATH always wins over the actual version relative to the codegen binary.